31-plotting_sample_size_biological_context_coverage.Rmd

```diff
@@ -0,0 +1,461 @@
+  p.title <- cowplot::ggdraw() + 
+    cowplot::draw_label(plot.title, fontface = "bold", 
+                        size = 20)
+  
+  # combine to get the final plot
+  final.plot <- cowplot::plot_grid(p.title, panels, ncol = 1, 
+                                   rel_heights = c(0.1, 1))
+  
+  # save 11"W x 5"H file (the type of file is determined by the file extension
+  # provided in output.file)
+  ggsave(output.file, plot = final.plot, width = 11, height = 5)
+  
+}
+```
+
+#### Directory setup
+
+```{r}
+# plot and result directory setup for this notebook
+plot.dir <- file.path("plots", "31")
+dir.create(plot.dir, recursive = TRUE, showWarnings = FALSE)
+results.dir <- file.path("results", "31")
+dir.create(results.dir, recursive = TRUE, showWarnings = FALSE)
+```
+
+## Read in data
+
+For both the biological context and sample size evaluations, the results we
+will be plotting are from `30-evaluate_sample_size_and_biological_context`.
+
+### Biological context
+
+```{r}
+# pathway coverage and proportion of LVs associated with pathways
+context.coverage.file <- file.path("results", "30", 
+                                   "biological_context_pathway_coverage.tsv")
+context.coverage.df <- readr::read_tsv(context.coverage.file) %>%
+  # clean up the context information for plotting
+  dplyr::mutate(biological_context = gsub("_", " ", biological_context)) %>%
+  # reorder by the sample size (low -> high)
+  dplyr::mutate(biological_context = factor(biological_context,
+                                            levels = c("blood", "cancer",
+                                                       "tissue", "cell line",
+                                                       "other tissues")))
+# number of latent variables
+context.num.file <- file.path("results", "30", 
+                              "biological_context_number_of_lvs.tsv")
+context.num.df <- readr::read_tsv(context.num.file) %>%
+  # clean up the context information for plotting
+  dplyr::mutate(biological_context = gsub("_", " ", biological_context)) %>%
+  # reorder by the sample size (low -> high)
+  dplyr::mutate(biological_context = factor(biological_context,
+                                            levels = c("blood", "cancer",
+                                                       "tissue", "cell line",
+                                                       "other tissues")))
+```
+
+### Sample size 
+
+```{r}
+# pathway coverage and proportion of LVs associated with pathways
+size.coverage.file <- file.path("results", "30", 
+                                "subsampled_pathway_coverage.tsv")
+# for plotting, we want the sample size as a factor (otherwise it's hard to
+# see what is happening on the lower end of things) but to order by the integer
+# value
+size.coverage.df <- readr::read_tsv(size.coverage.file) %>%
+    dplyr::mutate(sample_size = factor(sample_size,
+                                     levels = c("500", "1000", "2000", "4000",
```

Comment:
same thing here - it is nice to only change one thing if the levels need to change for some reason